### PR TITLE
Rename math operations

### PR DESCRIPTION
**Pull request by QGarchery**
Fixes cantina issue [#127](https://cantina.xyz/ai/8409a0ce-6c21-4cc9-8ef2-bd77ce7425af/findings/b399e353-f966-4657-81d6-2a12e043ad9f).

The only places where this differs from rounding towards negative infinity are the following:
1. https://github.com/morpho-org/morpho-blue-irm/blob/73eddf9e09731b75255b51d1ee14a321f5586b42/src/AdaptiveCurveIrm.sol#L123
2. https://github.com/morpho-org/morpho-blue-irm/blob/73eddf9e09731b75255b51d1ee14a321f5586b42/src/AdaptiveCurveIrm.sol#L137
3. the first computation `coeff.wMulDown(err)` in 
https://github.com/morpho-org/morpho-blue-irm/blob/73eddf9e09731b75255b51d1ee14a321f5586b42/src/AdaptiveCurveIrm.sol#L179

Those all have a consistent effect: under-estimating the error. This is why I think that only renaming those functions is enough

### CODE REVIEW

**Review by MerlinEgalite — APPROVED** [review]

File: src/libraries/MathLib.sol [hunk 1]
```diff
@@ -10,11 +10,13 @@ int256 constant WAD_INT = int256(WAD);
 /// @custom:contact security@morpho.org
 /// @notice Library to manage fixed-point arithmetic on signed integers.
 library MathLib {
-    function wMulDown(int256 a, int256 b) internal pure returns (int256) {
+    /// @dev Returns the multiplication of `a` by `b` (in WAD) rounded towards 0.
+    function wMul(int256 a, int256 b) internal pure returns (int256) {
```
Comment (Rubilmax) [inline src/libraries/MathLib.sol:14]:
```suggestion
    function wMulToZero(int256 a, int256 b) internal pure returns (int256) {
```

Comment (Rubilmax) [reply src/libraries/MathLib.sol:14]:
```suggestion
    function wMulTo0(int256 a, int256 b) internal pure returns (int256) {
```

Comment (Rubilmax) [reply src/libraries/MathLib.sol:14]:
```suggestion
    function wMulZero(int256 a, int256 b) internal pure returns (int256) {
```

Comment (Rubilmax) [reply src/libraries/MathLib.sol:14]:
```suggestion
    function wMul0(int256 a, int256 b) internal pure returns (int256) {
```

**Review by Rubilmax — APPROVED** [review]
I suggested alternatives, but I think it's fine having them named `wMul` and `wDiv` too

**Review by Jean-Grimal — APPROVED** [review]

**Review by adhusson — APPROVED** [review]
Agreed that current names are misleading. Not sure about the best new names. 
- just `wMul`(etc) makes sense because regular solidity regular division truncates.
- `wMul[zero,0,etc]` (etc) is more explicit and probably better since those new names will coexist with `wDivDown(uint)` (etc).

**Review by MerlinEgalite — APPROVED** [review]
formatting issue it seems

**Review by MerlinEgalite — APPROVED** [review]

**Comment by MerlinEgalite** [comment]
conflicts :/

**Review by Rubilmax — APPROVED** [review]

**Review by Rubilmax — APPROVED** [review]

**Comment by MerlinEgalite** [comment]
There are failing tests

**Review by MerlinEgalite — APPROVED** [review]

File: src/libraries/MathLib.sol [hunk 2]
```diff
@@ -10,11 +10,13 @@ int256 constant WAD_INT = int256(WAD);
 /// @custom:contact security@morpho.org
 /// @notice Library to manage fixed-point arithmetic on signed integers.
 library MathLib {
-    function wMulDown(int256 a, int256 b) internal pure returns (int256) {
+    /// @dev Returns the multiplication of `a` by `b` (in WAD) rounded towards 0.
+    function wMulTo0(int256 a, int256 b) internal pure returns (int256) {
```
Comment (StErMi) [inline src/libraries/MathLib.sol:14]: My two cents/nitpick: using the full word "zero" would maybe be more clean.

1) it's easy to get confused between char `0` and `o` and `O`. Having "zero" directly in the name is also much more explicit
2) probably it's not comfortable to have a number inside a function when you need to type it

Note: The solidity style guide does not have a strong opinion about having numbers inside the function's name

**Pull request merged**